File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(index, total, target,numbers):
    if index == len(numbers):
        if total == target:
            return 1
        else:
            return 0
        

    answer = 0
    answer += dfs(index+1, total + numbers[index], target, numbers)
    answer += dfs(index+1, total - numbers[index], target, numbers)
    
    return answer

    
    
def solution(numbers, target):
    answer = dfs(0, 0, target, numbers)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    
    
    const dfs = (cur_idx, sum) => {
        if (cur_idx === numbers.length) {
            if (sum === target) answer+=1
            return
        }
        
        // 더하기
        dfs(cur_idx+1, sum+numbers[cur_idx])
        // 빼기
        dfs(cur_idx+1, sum-numbers[cur_idx])
    }
    
    dfs(0,0)
    
    
    return answer;
}